File: src/main/java/es/uniovi/wifi/model/Localizacion.java
package es.uniovi.wifi.model;

import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import org.hibernate.validator.constraints.NotBlank;

@Entity
@Table(name = "TLocalizacion")
public class Localizacion {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id")
	private Long id;

	@NotBlank
	@Column(name = "nombre")
	private String nombre;

	@OneToMany(mappedBy = "localizacion")
	@JoinColumn(name = "id_localizacion", nullable = false)
	private Set<Wifi> wifis;

	public Localizacion() {
	}

	public Localizacion(String nombre) {
		this.nombre = nombre;
	}

	public Long getId() {
		return id;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public Set<Wifi> getWifis() {
		return wifis;
	}

	public void setWifis(Set<Wifi> wifis) {
		this.wifis = wifis;
	}

}

File: backend/src/main/kotlin/com/alberto/Intensidad/model/Localizacion.kt
package com.alberto.Intensidad.model

import javax.persistence.*
import javax.validation.constraints.NotBlank

@Entity
@Table(name = "TLocalizacion")
data class Localizacion (
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        val id: Long = 0,

        @NotBlank
        val nombre: String = "",

        @NotBlank
        val latitud: Double = 0.0,

        @NotBlank
        val longitud: Double = 0.0,

        @OneToMany(mappedBy = "localizacion")
        val intensidades: List<Intensidad> ?= null
)